packages/analytics-nextjs/src/hooks/useQueue.ts

```diff
@@ -13,24 +11,28 @@ export interface QueueMethods<T> {
     size: number;
 }
 
-export function useQueue<T>(initialValue: T[] = []): QueueMethods<T> {
-    const [state, set] = useState(initialValue);
+export function useQueue<T = any>(initialState: T[] = []): QueueMethods<T> {
```

Comment:
The original implementation from `react-use` was actually returning a whole new object each time, which cause a long chain of unwanted updates that ended up triggering the `page` tracking effect in an endless loop.